packages/extension-api/src/extension-api.d.ts

```diff
@@ -1953,13 +1961,9 @@ declare module '@podman-desktop/api' {
     export function stopContainer(engineId: string, id: string): Promise<void>;
     export function deleteContainer(engineId: string, id: string): Promise<void>;
     export function buildImage(
-      containerBuildContextDirectory: string,
-      relativeContainerfilePath: string,
-      imageName: string,
-      platform: string,
-      selectedProvider: ProviderContainerConnectionInfo | containerDesktopAPI.ContainerProviderConnection,
+      context: string,
       eventCollect: (eventName: 'stream' | 'error' | 'finish', data: string) => void,
-      abortController?: AbortController,
+      option?: BuildImageOptions,
```

Comment:
I have a remark

you designed option being an optional parameter but some of the parameters are mandatory (like the provider)

so probably the option should only contain the optional parameters (then we can omit it) or it's a mandatory parameter

```suggestion
      options?: BuildImageOptions,
```

Reply:
> you designed option being an optional parameter but some of the parameters are mandatory (like the provider)

Why does the provider should be mandatory ? 

Inside the container registry we can get the default one following our own logic (in favor of podman) `getFirstRunningConnection`. 

I feel like it is easier to make is optional, so extension can set it only if they have a requirements on a specific provider. Otherwise most of extension will simply get all providers and take the first one (as our implementation with @lstocchi ).

> so probably the option should only contain the optional parameters (then we can omit it) or it's a mandatory parameter

I agree that the option should only be the optional elements, that's why the first arguments in the buildImage is the context dir, from a user point of view the "podman build ." is the simplest method it can use.